src/frameworks/ccpa_from_us_privacy_string.js

```diff
@@ -0,0 +1,47 @@
+const FrameworkBase = require('./base');
+
+/**
+ * Implements usprivacy string framework
+ * for more information on the US Privacy string see:
+ * https://github.com/InteractiveAdvertisingBureau/USPrivacy/blob/master/CCPA/US%20Privacy%20String.md#us-privacy-string-format
+ */
+class CCPAFromUSPrivacyString extends FrameworkBase {
+  constructor() {
+    super();
+    this.usPrivacyString = '';
+  }
+
+  useConfig({ usp }) {
+    if (usp) {
+      this.usPrivacyString = ('' + usp).toUpperCase();
+    }
+  }
+
+  supportedCapabilities() {
+    return ['canUsePersonalInformationForTargeting', 'hasBeenNotifiedOfRights'];
+  }
+
+  canUsePersonalInformationForTargeting() {
+    return this.consentStringAllowsPersonalDataSale();
+  }
+
+  hasBeenNotifiedOfRights() {
+    return this.consentStringAcknowledgesUserHasBeenNotifiedOfRights();
+  }
+
+  consentStringAllowsPersonalDataSale() {
+    if (!this.supportedUsPrivacyStringVersion()) return true;
```

Comment:
Is it right to _fail true_ if there *is* a USP string, but it's ill-formed, or (someday) a different version?

Reply:
For CCPA, I believe that is true, since it defaults ("fails") to be permissive. ~We~ The legislation need a very clear sign to *block* the sale of user data.